pkg/kstatus/polling/engine/engine.go

```diff
@@ -85,15 +84,35 @@ func (s *PollerEngine) Poll(ctx context.Context, identifiers []wait.ResourceIden
 			previousResourceStatuses: make(map[wait.ResourceIdentifier]*event.ResourceStatus),
 			eventChannel:             eventChannel,
 			statusAggregator:         aggregator,
-			pollUntilCancelled:       pollUntilCancelled,
-			pollingInterval:          pollInterval,
+			pollUntilCancelled:       options.PollUntilCancelled,
+			pollingInterval:          options.PollInterval,
 		}
 		runner.Run()
 	}()
 
 	return eventChannel
 }
 
+// Options contains the different parameters that can be used to adjust the
+// behavior of the PollerEngine.
+type Options struct {
+	// PollUntilCancelled defines whether the engine should stop polling and close the
+	// event channel when the Aggregator implementation considers all resources to have reached
+	// the desired status.
+	PollUntilCancelled bool
+
+	// PollInterval defines how often the PollerEngine should poll the cluster for the latest
+	// state of the resources.
+	PollInterval time.Duration
+
+	// WaitForDeletion defines which implementation of the aggregator interface
+	// that should be used. If the value is false, then the aggregator will consider
+	// the wait completed when all resources have reached the Current status. If
+	// the value it true, it will stop polling and return when all resources
+	// have reached the NotFound status
+	WaitForDeletion bool
```

Comment:
Can we express PollUntilCancelled and WaitForDeletion as distinct items in an enum to eliminate nonsensical options (e.g. both true)?